### PR TITLE
redis-desktop-manager: 0.9.1 -> 2022.5, rename to RESP.app

### PR DESCRIPTION
**Pull request by teutat3s**
###### Description of changes

ZHF: #199919

fixes https://hydra.nixos.org/build/197417396/nixlog/1

RedisDesktopManager is now RESP.app - GUI for Redis: quote from upstream "RESP.app is joining forces with Redis to offer the Redis community the best possible developer experience"

For details, see:
https://github.com/RedisInsight/RedisDesktopManager
and
https://redis.com/blog/respapp-joining-redis/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/resp-app/default.nix [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      "''${qtWrapperArgs[@]}"
+  '';
+
+  meta = with lib; {
+    broken = (stdenv.isLinux && stdenv.isAarch64);
+    description = "Cross-platform Developer GUI for Redis";
+    homepage = "https://resp.app/";
+    license = licenses.gpl3Only;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ cstrahan ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/resp-app/default.nix:114]:
```suggestion
    description = "Cross-platform Developer GUI for Redis";
    homepage = "https://resp.app/";
    license = licenses.gpl3Only;
    platforms = platforms.linux;
    maintainers = with maintainers; [ cstrahan ];
    broken = stdenv.isLinux && stdenv.isAarch64;
```
Is it still broken on aarch64-linux?

File: pkgs/applications/misc/resp-app/default.nix [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    "SYSTEM_BROTLI=1"
+    "VERSION=${version}"
+    "src/resp.pro"
+  ];
+
+  dontWrapQtApps = true;
+
+  postInstall = ''
+    buildPythonPath "$out $pythonPath"
+    wrapProgram $out/bin/resp \
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/resp-app/default.nix:103]: Isn't this double wrapping?

File: pkgs/applications/misc/resp-app/default.nix [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  patchPhase = ''
+    substituteInPlace src/resp.pro \
+      --replace 'which ccache' "false"
+    substituteInPlace src/resp.pro \
+      --replace 'target.files = $$DESTDIR/resp' "${placeholder "src"}/bin/linux/release/resp"
+    substituteInPlace src/resp.pro \
+      --replace '/opt/resp_app' "${placeholder "out"}"
+    substituteInPlace src/resp.pro \
+      --replace 'target.path = $$LINUX_INSTALL_PATH' 'target.path = $$LINUX_INSTALL_PATH/bin'
+    substituteInPlace src/resp.pro \
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/resp-app/default.nix:86]:
```suggestion
      --replace 'which ccache' "false" \
      --replace 'target.files = $$DESTDIR/resp' "${placeholder "src"}/bin/linux/release/resp" \
      --replace '/opt/resp_app' "${placeholder "out"}" \
      --replace 'target.path = $$LINUX_INSTALL_PATH' 'target.path = $$LINUX_INSTALL_PATH/bin' \
```

File: pkgs/applications/misc/resp-app/default.nix [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  pythonPath = with python3Packages; [
+    bitstring
+    cbor
+    msgpack
+    phpserialize
+    redis-rdb-tools
+    python-lzf
+  ];
+
+  patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/resp-app/default.nix:77]:
```suggestion
  postPatch = ''
```

File: pkgs/applications/misc/resp-app/default.nix [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+, qttools
+, snappy
+, wrapQtAppsHook
+, zstd
+}:
+
+let
+  redis-rdb-tools = with python3Packages; buildPythonPackage rec {
+    pname = "redis-rdb-tools";
+    version = "resp_rdbtools-0.1.15";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/resp-app/default.nix:28]:
```suggestion
    version = "0.1.15";
```
Can we move this to pythonPackages?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/resp-app/default.nix (hunk 2 above)
Comment (teutat3s) [reply pkgs/applications/misc/resp-app/default.nix:103]: Without this double wrapping, I couldn't get our packaged `pyotherside` into the wrapper and saw this error:
```
"PyOtherSide error: Traceback (most recent call last):\n\n  File \"qrc:/python//formatters/__init__.py\", line 1, in <module>\n    from .binary import BinaryFormatter\n\n  File \"qrc:/python//formatters/binary.py\", line 1, in <module>\n    import bitstring\n\nModuleNotFoundError: No module named 'bitstring'\n"
```

Comment (teutat3s) [reply pkgs/applications/misc/resp-app/default.nix:103]: Got it to work using
```
  preFixup = ''
    buildPythonPath "$pythonPath"
    qtWrapperArgs+=(--prefix PYTHONPATH : "$program_PYTHONPATH")
  '';
```

File: pkgs/applications/misc/resp-app/default.nix (hunk 5 above)
Comment (teutat3s) [reply pkgs/applications/misc/resp-app/default.nix:28]: Found out it's already packaged as `rdbtools` and only needs a patch to work with `RESP.app`

File: pkgs/applications/misc/resp-app/default.nix (hunk 1 above)
Comment (teutat3s) [reply pkgs/applications/misc/resp-app/default.nix:114]: Let's find it out :smile:

**Comment by teutat3s** [comment]
@SuperSandro2000 ready for another review, whenever you spare a minute.

**Comment by SuperSandro2000** [comment]
@ofborg build resp-app

**Pull request merged**